app/script/event/EventRepository.js

```diff
@@ -672,7 +673,8 @@ z.event.EventRepository = class EventRepository {
 
       const mappedIsMessageAdd = mappedType === z.event.Client.CONVERSATION.MESSAGE_ADD;
       const storedIsMessageAdd = storedType === z.event.Client.CONVERSATION.MESSAGE_ADD;
-      const userReusedId = !mappedIsMessageAdd || !storedIsMessageAdd || !mappedData.previews.length;
+      const userReusedId =
+        !mappedIsMessageAdd || !storedIsMessageAdd || !(mappedData.previews.length || mappedData.replacing_message_id);
```

Comment:
Let's explain what we filter for in `!(mappedData.previews.length || mappedData.replacing_message_id)` by pulling it out.